apps/auth/nextjs-ssr/src/hooks/auth.ts

```diff
@@ -0,0 +1,29 @@
+import { cookies } from "next/headers";
+import { createCrossmint, CrossmintAuth } from "@crossmint/server-sdk";
+
+export async function getAuthSession(refreshRoute?: string) {
+    const cookieStore = await cookies();
+    const jwt = cookieStore.get("crossmint-jwt")?.value;
+    const refreshToken = cookieStore.get("crossmint-refresh-token")?.value;
+
+    if (!refreshToken) {
+        return;
+    }
+
+    try {
+        const crossmint = createCrossmint({
+            apiKey: process.env.SERVER_CROSSMINT_API_KEY || "",
+        });
+        const crossmintAuth = CrossmintAuth.from(crossmint, {
+            refreshRoute,
+        });
+
+        const session = await crossmintAuth.getSession({
```

Comment:
this doesnt refresh the token, correct?

Reply:
Nope, if the jwt is valid, there's no need to refresh. And the refresh had just happened in the middleware so we're good